src/Agent/NewRelic/Profiler/Configuration/Configuration.h

```diff
@@ -605,13 +605,23 @@ namespace NewRelic { namespace Profiler { namespace Configuration {
                 return false;
             }
 
-            if (IsAzureFunction()) {
-                auto retVal = ShouldInstrumentAzureFunction(processPath, appPoolId, commandLine);
-                if (retVal == 0) {
-                    return false;
+            // returns -1 if env var isn't specified at all
+            auto isAzureFunctionModeEnabled = IsAzureFunctionModeEnabled();
+
+            if (IsAzureFunction())
+            {
+                if (isAzureFunctionModeEnabled == 1) {
+                    auto retVal = ShouldInstrumentAzureFunction(processPath, appPoolId, commandLine);
+                    if (retVal == 0) {
+                        return false;
+                    }
+                    if (retVal == 1) {
+                        return true;
+                    }
                 }
-                if (retVal == 1) {
-                    return true;
+                else if (isAzureFunctionModeEnabled == 0)
+                {
+                    return false;
```

Comment:
Is this the behavior we want? Or should this be the same behavior as if the environment variable is not present?

Reply:
If the variable is present with a value of 0, the user has (presumably) explicitly said they don't want to instrument Azure functions -- in the current preview, the user generally won't know about the environment variable and thus, it won't exist. So when it *does* exist, we need to respect it. 

I'm really positioning this bit to remain functional when instrumentation is enabled by default.

Reply:
I'm mostly calling this out because with the current implementation, the setting acts more like an enumeration than a boolean because it has 3 states. Most of our boolean settings are either on or off, and if the setting is missing we default it to either on or or off. In this case we have:

1. Missing - do not perform the process check
2. On - Perform the process check
3. Off - Never instrument anything in an azure function environment

This behavior is different than the behavior we have within the managed agent, where I think Off and missing have the same behavior.